worker/README.md

```diff
@@ -1,5 +1,33 @@
 # worker
 
+*worker - container orchestrator/ worker artifact management/ build executor*
+
 ![](https://upload.wikimedia.org/wikipedia/commons/thumb/b/b8/Otto_Lilienthal_gliding_experiment_ppmsca.02546.jpg/640px-Otto_Lilienthal_gliding_experiment_ppmsca.02546.jpg)
 
 [from](https://commons.wikimedia.org/wiki/File:Otto_Lilienthal_gliding_experiment_ppmsca.02546.jpg) the [United States Library of Congress's](https://www.loc.gov/) [Prints and Photographs division](https://www.loc.gov/rr/print/)
+
+
+
+
+
+
+  ## about
+
+  *worker* is the workhorse of Concourse. It's responsible for the creation
+  and deletion of the containers in which the pipeline tasks are executed.
+
+  A worker node registers with the web node and is then used for executing
+  builds and performing resource checks. It doesn't really decide much on its own.
```

Comment:
We can elaborate a bit more on `It doesn't really decide much on its own` - as a developer reading this, I'd be asking myself the question, "if the worker doesn't make decisions, then who does?" and the answer is that all container allocation logic happens in the ATC.

Perhaps one way to add a bit more detail is to say instead, "...performing resource checks. The ATC component in the web node(s) decides how to allocate containers to workers that have been registered in the pool, using the configured `container-placement-strategy`".

Reply:
I like those suggestions 👍 Should we also mention that ATC also handles container deletion as well? Re-reading this it sounds as though that the worker take care of container cleanup themselves when its more a GC step? Curious on your thoughts

Reply:
Sure, happy for you to include mention of container deletion, that feels like a logical thing to include.

For more context (the below is probably too much detail to include in this level of documentation), it's a bit confusing because we're not as precise as we could be when we talk about "containers" and their lifecycle. You're absolutely right that the ATC initiates container deletion in the sense that the ATC will do the following things:

1. add/update a container record in its database as "creating"/"deleting"
2. send an API call along to the container's worker's Garden server asking, "please create/delete this container!"
3. update container record to "created", or delete the row in the case of deletion

But on the worker, the Garden server is the thing that actually gets rid of the container as a filesystem entity, when it hears the request from the ATC. So _technically_.. the last mile of deletion and creation is performed by something on the worker, but that's kind of an implementation detail that we shouldn't worry about too much when we're describing the responsibilities of components in the system. (Apologies if you already knew this context - I couldn't find a clear explanation anywhere in our [existing written documentation](https://concourse-ci.org/container-internals.html) so I figured I'd type it out!)